### DISCUSSION TITLE
Do we have any module similar to "performIoHidEvent" available for Android platform?

### DISCUSSION THREAD

**Discussion by pralaykdas** [appium/appium#17023] [Q&A]
I want to send the events to an android device using the [HID Keyboard ](https://source.android.com/devices/input/keyboard-devices#hid-keyboard-and-keypad-page-0x07) 

For iOS, we can use the following module:
_driver.execute_script(“mobile: performIoHidEvent”, {“page”: 0x07, “usage”: 0x14, “durationSeconds”: 0.5})_

So, do we have any similar modules for the Android platform?

**Comment by jainilparikh** [comment]
@mykola-mokhnach  could you please help us out here?

We want to mimic the keypress event as discussed [here](https://github.com/appium/appium/issues/16445)

Your solution for iOS using "performIoHidEvent" was exactly what we needed.

Can you let us know if there is a similar way to send keyboard events(using HID maybe) to Android as well?

**Comment by mykola-mokhnach** [comment]
On Android there is [pressKey](https://github.com/appium/java-client/blob/0e05078d0ae8e4bebb80080790712e933e2ea666/src/test/java/io/appium/java_client/android/KeyCodeTest.java#L58) API
